c/tree.c

```diff
@@ -808,11 +985,16 @@ LOCAL MMDBW_status insert_record_into_current_record(
     MMDBW_record_s *current_record,
     MMDBW_network_s *network,
     MMDBW_record_s *new_record,
-    MMDBW_merge_strategy merge_strategy,
-    bool is_internal_insert)
-{
-    if (current_record->type == MMDBW_RECORD_TYPE_FIXED_NODE) {
-        return MMDBW_FIXED_NODE_OVERWRITE_ATTEMPT_ERROR;
+    MMDBW_merge_strategy merge_strategy)
+{
+    // We only get called when we have a current_record with these record
+    // types. There was previously logic for other types, but that was
+    // confusing.
+    if (current_record->type != MMDBW_RECORD_TYPE_EMPTY &&
+        current_record->type != MMDBW_RECORD_TYPE_DATA &&
+        current_record->type != MMDBW_RECORD_TYPE_NODE) {
+        croak(
```

Comment:
I've been trying to move away from using `croak`s in this code as it makes it even harder to keep track of what needs to be freed or cleaned up. However, based on the description, this should only happen if there is a programming error in the writer, which seems fine.

Reply:
I see, yeah that makes sense. Yes, this one should probably not happen in normal use either, although I think it would be easy to just use an error status. I'll take a look at it and others I added.